Add generation of "retry regressed" list

LukasKalbertodt opened the pull request:
This is a list that can be passed to crater as "crates" argument. Currently, there is no easy way to obtain such a list from a report.

Just let me know if I should change something!

lqd commented:
cc @ecstatic-morse 🎉

pietroalbini commented:
I'll review in a couple of days max!

pietroalbini commented:
I haven't forgotten about this, I'm just busy with university.

LukasKalbertodt commented:
@pietroalbini Don't worry, I know that situation very well ^_^

pietroalbini commented on src/report/mod.rs:
```diff
@@ -257,6 +259,11 @@ pub fn gen<DB: ReadResults, W: ReportWriter + Display>(
         &mime::APPLICATION_JSON,
     )?;
+    dest.write_string(
+        "retry_regressed_list.txt",
```
Can we rename the file `retry-regressed-list.txt`?

pietroalbini commented on src/report/mod.rs:
```diff
@@ -725,8 +760,34 @@ mod tests {
             EncodedLog::Plain(b"beta log".to_vec()),
         );
 
+        // db.add_dummy_sha(&ex, reg.clone(), "f01".to_string());
```
Commented line?

LukasKalbertodt commented:
@pietroalbini No problem! Both issues fixed.

pietroalbini commented:
Thanks!

@bors r+

bors commented:
:pushpin: Commit 38a4100ed9f6ce00989003711c33bd75d809fcf4 has been approved by `pietroalbini`

<!-- @bors r=pietroalbini 38a4100ed9f6ce00989003711c33bd75d809fcf4 -->
<!-- homu: {"type":"Approved","sha":"38a4100ed9f6ce00989003711c33bd75d809fcf4","approver":"pietroalbini"} -->

bors commented:
:hourglass: Testing commit 38a4100ed9f6ce00989003711c33bd75d809fcf4 with merge ad19827dc652c169f9a7c68e837f92254b761bd1...
<!-- homu: {"type":"BuildStarted","head_sha":"38a4100ed9f6ce00989003711c33bd75d809fcf4","merge_sha":"ad19827dc652c169f9a7c68e837f92254b761bd1"} -->

bors commented:
:broken_heart: Test failed - [checks-actions](https://help.github.com/en/actions)
<!-- homu: {"type":"BuildFailed","builder_url":"https://help.github.com/en/actions","builder_name":"checks-actions"} -->

LukasKalbertodt commented:
Oh.. by adding `krate` to `CrateResult`, I changed the JSON output of all tests. Shall I simply bless all the minicrater tests or do you have a better idea?

pietroalbini commented:
Just bless all the tests.

LukasKalbertodt commented:
Rebased, blessed tests and fixed [`array_into_iter` lint](https://github.com/rust-lang/rust/issues/66145). With the last one, I've gone full circle, since I was only working with crater to get `[T; N]::into_iter` land :grin:

pietroalbini commented:
@bors r+

bors commented:
:pushpin: Commit 4a8ecbfcce03055c46ba91af27deeb6ddcb69a79 has been approved by `pietroalbini`

<!-- @bors r=pietroalbini 4a8ecbfcce03055c46ba91af27deeb6ddcb69a79 -->
<!-- homu: {"type":"Approved","sha":"4a8ecbfcce03055c46ba91af27deeb6ddcb69a79","approver":"pietroalbini"} -->

bors commented:
:hourglass: Testing commit 4a8ecbfcce03055c46ba91af27deeb6ddcb69a79 with merge 29add6e19ef7cf67aa2aab12fcf2e44dfc673bc3...
<!-- homu: {"type":"BuildStarted","head_sha":"4a8ecbfcce03055c46ba91af27deeb6ddcb69a79","merge_sha":"29add6e19ef7cf67aa2aab12fcf2e44dfc673bc3"} -->

bors commented:
:sunny: Test successful - [checks-actions](https://help.github.com/en/actions)
Approved by: pietroalbini
Pushing 29add6e19ef7cf67aa2aab12fcf2e44dfc673bc3 to master...
<!-- homu: {"type":"BuildCompleted","approved_by":"pietroalbini","base_ref":"master","builders":{"checks-actions":"https://help.github.com/en/actions"},"merge_sha":"29add6e19ef7cf67aa2aab12fcf2e44dfc673bc3"} -->
